### ISSUE TITLE
[BUG]: AWS sdk v3 causing methods to throw error

### ISSUE THREAD

**Issue by matthelliwell2** [sensedeep/dynamodb-onetable#89]
**Describe the bug**
If you don't pass a client into the table constructor and instead use the setClient method then operations throw an exception:

```

this.table.client[DocumentClientMethods[op]](...).promise is not a function
TypeError: this.table.client[DocumentClientMethods[op]](...).promise is not a function
    at Model.<anonymous> (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:346:90)
    at Generator.next (<anonymous>)
    at C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:8:71
    at new Promise (<anonymous>)
    at Object.<anonymous>.__awaiter (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:4:12)
    at Model.run (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:279:16)
    at Model.<anonymous> (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:674:31)
    at Generator.next (<anonymous>)
    at C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:8:71
    at new Promise (<anonymous>)
```

**To Reproduce**

Create a table without a client and set the client using setClient
```
export const accountsTable = new Table({ name: 'mytable', delimiter: '#', timestamps: true, isoDates: true, schema: schema })
accountsTable.setClient(client)
```

Operations on the table such as create now throw the exception above. This looks like it is due to the setClient method not setting the V3 flag.

**Expected behavior**
Database operations should suceed

**Environment (please complete the following information):**
 - OS windows 10
 - Node Version 14

**Additional context**

**Comment by mobsense** [comment]
Hmmm,

We have such a scenario in our test suite and it seems to work for us. test/high-level-table-apis.ts

The setClient code does:

```
    setClient(client) {
        this.client = client
        this.V3 = client.V3
        this.service = this.V3 ? this.client : this.client.service
    }
```

So this should propagate the client V3.

Note that the client in the V3 case must be an instance of the Dynamo helper:

```
const Client = new Dynamo({
    client: new DynamoDBClient({
        endpoint: `http://localhost:${PORT}`,
        region: 'local',
        credentials: new AWS.Credentials({
            accessKeyId: 'test',
            secretAccessKey: 'test',
        }),
    })
})
```

It is not a raw AWS client.

**Comment by matthelliwell2** [comment]
I'll go back and debug what's going on in my case but I agree the setClient code looks like its doing what it should.

**Comment by matthelliwell2** [comment]
I think I've reproduced the problem. Calling the create method on the code below throws the error mentioned:

```
import { OneSchema, Table } from 'dynamodb-onetable'
import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
import { DynamoDBClient } from '@aws-sdk/client-dynamodb'
import Dynamo from 'dynamodb-onetable/Dynamo'

const schema: OneSchema = {
    indexes: { primary: { hash: 'pk' } },
    models: {
        Bordereaux: {
            pk: {
                type: String,
                required: true,
                hidden: false
            },
            fileName: { type: String, required: true }
        }
    }
}

const client = new DynamoDBClient({
    region: 'eu-west-1'
})

const docClient = DynamoDBDocument.from(client)
const oneTableClient = new Dynamo({ client: docClient })

export const table = new Table({ name: 'TestTable', delimiter: '#', timestamps: true, isoDates: true, schema: schema })

const Bordereaux = table.getModel('Bordereaux')

export async function create() {
    table.setClient(oneTableClient)
    return Bordereaux.create({ fileName: 'file1', pk: 'file1#' })
}

```

The problem is that the V3 flag is set on the model in the model constructor and is a copy of the v3 flag on the table. Because I haven't set a client on the table when the model is created, the v3 flag is false. I think that when table.setClient is called, it needs to update the v3 flag on the models.

If I add the parameter ```client: {V3: true}``` to the table constructor then it gets past this problem.

**Comment by mobsense** [comment]
Thank you for nailing this down.

The issue is Model doing 

    this.V3 = table.V3

Fix coming.

**Comment by mobsense** [comment]
Fixed in 1.6.4
